client/data/agency-dashboard/use-fetch-dashboard-sites.ts

```diff
@@ -55,7 +57,16 @@ const useFetchDashboardSites = (
 		{
 			select: ( data ) => {
 				return {
-					sites: data.sites,
+					sites: data.sites.map( ( site: Site ) => {
+						const data: { connected: boolean } | undefined = queryClient.getQueryData( [
+							'jetpack-agency-test-connection',
+							site.blog_id,
+						] );
+						return {
+							...site,
+							is_connected: data ? data.connected : true,
+						};
+					} ),
```

Comment:
We're retrieving the data from the cache initialized by the API request to the `/test-connection` endpoint. When we execute this callback function, is there a guarantee that we will always have the data available (for every site)?

Reply:
If that site's`/test-connection` API is already called, it'll always be available since we are caching it for 5 min. 

This is why we are updating it at both the APIs - `/test-connection` and `/sites` so that we always have the latest connection status.